src/main/java/hq/CaseAPIs.java

```diff
@@ -51,4 +57,28 @@ public static String filterCases(CaseFilterRequestBean request) throws Exception
         }
         return "null";
     }
+
+    public static CaseBean getFullCase(String caseId, String username){
+        UserSqlSandbox mSandbox = new UserSqlSandbox(username);
+        SqliteIndexedStorageUtility<Case> caseStorage = mSandbox.getCaseStorage();
+        Case cCase = caseStorage.getRecordForValue("case_id", caseId);
+        CaseBean ret = new CaseBean(cCase);
+        return ret;
```

Comment:
Seems like quite a lot of work to be doing inside a tight loop especially since the 1st 2 lines are always going to be the same for each loop iteration (in `filterCasesFull`)